application/controllers/Hook.php

```diff
@@ -0,0 +1,254 @@
+            log_message('error', "Could not decode JSON data from the file");
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("unable to decode JSON data");
+        }
+
+        //retrieving version number
+        preg_match($this->pattern, $filename, $matches);
+        if (!isset($matches[1])) {
+            exit("could not retrieve version from filename '$filename'");
+        }
+        $version = $matches[1];
+        if (strlen($matches[1]) < 1) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("version found not correct ('$version') from filename '$filename'");
+        }
+
+        //starting real stuff
+        //create the execution
+        $stats = $file_contents->stats;
+        $execution_data = [
+            'ref' => date('YmdHis'),
+            'start_date' => date('Y-m-d H:i:s', strtotime($stats->start)),
+            'end_date' => date('Y-m-d H:i:s', strtotime($stats->end)),
+            'duration' => $stats->duration,
+            'version' => $version
+        ];
+        //let's check if there's not a similar entry...
+        $entry_date = date('Y-m-d', strtotime($stats->start));
+        $similar = $this->Execution->findSimilarEntries($entry_date, $version);
+
+        if ($similar !== NULL) {
+            if (!$this->force) {
+                log_message('error', "A similar entry was found (criteria: version $version and date $entry_date)");
+                header($_SERVER['SERVER_PROTOCOL'] . ' 409 Conflict', true, 409);
+                exit("A similar entry was found (criteria: version '$version' and date '$entry_date'). Use the 'force' parameter to force insert");
+            } else {
+                log_message('warning', "A similar entry was found (criteria: version $version and date $entry_date) but FORCING insert anyway");
+            }
+        }
+
+        try {
+            log_message('info', "Inserting Execution");
+            $this->execution_id = $this->Execution->insert($execution_data);
+        } catch(Exception $e) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 500 Internal Server Error', true, 500);
+            exit("could not insert execution");
+        }
+
+        //launching into orbit
+        $this->loopThrough($file_contents->suites);
+
+        //get the data from the database itself
+        $updated_data = $this->Execution->getSummaryData($this->execution_id);
+
+        $update_data = [
+            'skipped' => $updated_data->skipped,
+            'suites' => $updated_data->suites,
+            'tests' => $updated_data->tests,
+            'passes' => $updated_data->passed,
+            'failures' => $updated_data->failed,
+            'insertion_end_date' => "NOW()"
+        ];
+        //update the execution row with updated data
+        $this->Execution->update($update_data, $this->execution_id);
+        header($_SERVER['SERVER_PROTOCOL'] . ' 200 OK', true, 200);
```

Comment:
You could create a reusable private function for this, dont you think ?

Reply:
Thing is, how many different headers do I need ? I could use a function but idk if it's really worth it. 
What do you think ? Is it worth it for 3 different headers ?

Reply:
I did it.